dask/dataframe/core.py

```diff
@@ -363,6 +445,23 @@ def _rebuild(self, dsk, *, rename=None):
     def _constructor(self):
         return new_dd_object
 
+    @property
+    def _divisions(self):
+        warnings.warn(
+            "_divisions property is deprecated. Use divisions instead.",
+            FutureWarning,
+        )
+        return self.divisions
+
+    @_divisions.setter
+    def _divisions(self, value):
+        warnings.warn(
+            "Settable _divisions property is deprecated. "
+            "Set divisions attribute directly instead.",
+            FutureWarning,
+        )
+        self.divisions = value
```

Comment:
Do we want to define `divisions` and `_divisions` at the collection level? Im choosing to deprecate `_divisions` to simplify things a bit. Though, it does feel strange to encourage users to set `.divisions` directly...

Reply:
Generally I'd vote for just going with `divisions`. Do we have a read on how much usage the divisions setter gets? I could be totally wrong here but should people typically be using repartition instead of directly setting divisions?